nativeunwind/elfunwindinfo/elfgopclntab.go

```diff
@@ -332,6 +327,38 @@ func SearchGoPclntab(ef *pfelf.File) ([]byte, error) {
 	return nil, nil
 }
 
+// Determine the default unwinding strategy for this executable
+func (ee *elfExtractor) determineDefaultStrategy(arch elf.Machine, magic uint32) int {
+	if arch != elf.EM_AARCH64 {
+		return strategyFramePointer
+	}
+
+	// On ARM64, we may or may not be able to use frame pointers. Go 1.20 and
+	// earlier did not maintain frame pointers properly when the Go runtime
+	// copies the stack during `runtime.morestack` calls: all old frame pointers
+	// are set to 0.
+	//
+	// This was fixed for Go 1.21 and later in:
+	// https://github.com/golang/go/commit/a41a29ad19c25c3475a65b7265fcad870d954c2a
+	//
+	// We thus need to unwind with stack delta offsets for Go 1.20 and earlier
+	// but can use frame pointer unwinding on Go 1.21 and later.
+	if magic == magicGo1_2 || magic == magicGo1_16 || magic == magicGo1_18 {
+		return strategyDeltas
+	}
+	if magic != magicGo1_20 {
+		return strategyFramePointer
+	}
+
+	// The gopclntab magicGo1_20 not reveal exact version. This is the ambiguous
+	// case where and we need to determine if running Go 1.21 or later with this magic.
+	bi, err := buildinfo.Read(ee.file.FileReader())
```

Comment:
This shall use the `pfelf` GetGoVersion or similar helper once it becomes available.